subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -287,21 +287,30 @@ public void execute(SourceSet sourceSet) {
     private void configureJavaDoc(JavaPluginConvention pluginConvention) {
         Project project = pluginConvention.getProject();
 
-        SourceSet mainSourceSet = pluginConvention.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
-        Javadoc javadoc = project.getTasks().create(JAVADOC_TASK_NAME, Javadoc.class);
-        javadoc.setDescription("Generates Javadoc API documentation for the main source code.");
-        javadoc.setGroup(JavaBasePlugin.DOCUMENTATION_GROUP);
-        javadoc.setClasspath(mainSourceSet.getOutput().plus(mainSourceSet.getCompileClasspath()));
-        javadoc.setSource(mainSourceSet.getAllJava());
-        addDependsOnTaskInOtherProjects(javadoc, true, JAVADOC_TASK_NAME, COMPILE_CONFIGURATION_NAME);
-    }
+        final SourceSet mainSourceSet = pluginConvention.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
```

Comment:
nitpick: Seems like `pluginConvention` should be final here and then we can move the sourceSet lookup into the javadoc config action and resolve it later.  Then we would remove the ordering requirement between `configureSourceSets` and `configureJavaDoc`.